benchmark/NQueens.hs

```diff
@@ -0,0 +1,59 @@
+{-# LANGUAGE DeriveFunctor, FlexibleContexts, FlexibleInstances, LambdaCase, MultiParamTypeClasses, RankNTypes,
+             TypeApplications, TypeOperators, UndecidableInstances #-}
+{-# OPTIONS_GHC -Wno-redundant-constraints #-}
+
+-- Based largely on the implementation by Sreekar Shastry,
+-- available at https://github.com/sshastry/queenslogic
+
+module NQueens (runQueens, toplevel) where
```

Comment:
Huge 👍 to splitting off a new module for these benchmarks. I’d like to organize benchmarks by the effect(s) whose carriers we’re comparing, so what do you think about renaming the module to `NonDet` or `NonDet.NQueens`?